frontend/src/rauta.tsx

```diff
@@ -93,8 +93,29 @@ export type RouterLib = {
     Router: (props: RouterProps) => JSX.Element;
 };
 
-export type Listener = () => void;
-export type BeforeNavigationListener = (preventNavigation: () => void) => void;
+/** Helper class: a list of listeners */
+class Listeners<F extends (...args: any) => any> {
```

Comment:
I'm not sure about this type. This allows for heterogeneous lists.

Why not parameterize using the argument and return type directly?

Or maybe you can use `infer` here somehow ... :thinking: